pilot/cmd/pilot-agent/main.go

```diff
@@ -549,20 +489,8 @@ func detectSds(controlPlaneBootstrap bool, sdsAddress, jwtPath string) (bool, st
 		return true, jwtPath
 	}
 
-	if !controlPlaneBootstrap {
-		// workload sidecar
-		// treat sds as disabled if uds path isn't set.
-		if _, err := os.Stat(udsPath); err != nil {
-			return false, ""
-		}
-
-		return true, jwtPath
-	}
-
-	// controlplane components like pilot/mixer/galley have sidecar
-	// they start almost same time as sds server; wait since there is a chance
-	// when pilot-agent start, the uds file doesn't exist.
-	if !waitForFile(udsPath, sdsUdsWaitTimeout) {
```

Comment:
Why was this wait removed? Is it no longer needed?

Reply:
We no longer need to wait because its not created by nodeagent, we are creating it ourselves.